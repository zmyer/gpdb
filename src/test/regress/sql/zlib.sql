DROP TABLE IF EXISTS test_zlib_hashjoin;
CREATE TABLE test_zlib_hashjoin (i1 int, i2 int, i3 int, i4 int, i5 int, i6 int, i7 int, i8 int) WITH (APPENDONLY=true) DISTRIBUTED BY (i1) ; 
INSERT INTO test_zlib_hashjoin SELECT i,i,i,i,i,i,i,i FROM 
	(select generate_series(1, nsegments * 333333) as i from 
	(select count(*) as nsegments from gp_segment_configuration where role='p' and content >= 0) foo) bar;

SET gp_workfile_type_hashjoin=bfz;
SET gp_workfile_compress_algorithm=zlib;
SET statement_mem=5000;

--Fail after workfile creation and before add it to workfile set
\! gpfaultinjector -f workfile_creation_failure -y reset --seg_dbid 2
\! gpfaultinjector -f workfile_creation_failure -y error --seg_dbid 2

SELECT COUNT(t1.*) FROM test_zlib_hashjoin AS t1, test_zlib_hashjoin AS t2 WHERE t1.i1=t2.i2;

\! gpfaultinjector -f workfile_creation_failure -y status --seg_dbid 2

RESET statement_mem;
DROP TABLE IF EXISTS test_zlib_hagg; 
CREATE TABLE test_zlib_hagg (i1 int, i2 int, i3 int, i4 int);
INSERT INTO test_zlib_hagg SELECT i,i,i,i FROM 
	(select generate_series(1, nsegments * 300000) as i from 
	(select count(*) as nsegments from gp_segment_configuration where role='p' and content >= 0) foo) bar;

SET statement_mem=2000;

--Fail after workfile creation and before add it to workfile set
\! gpfaultinjector -f workfile_creation_failure -y reset --seg_dbid 2
\! gpfaultinjector -f workfile_creation_failure -y error --seg_dbid 2

SELECT MAX(i1) FROM test_zlib_hagg GROUP BY i2;

\! gpfaultinjector -f workfile_creation_failure -y status --seg_dbid 2

-- Reset faultinjectors
\! gpfaultinjector -f workfile_creation_failure -y reset --seg_dbid 2

create table t (i int, j text);
insert into t select i, i from generate_series(1,1000000) as i;
create table t1(i int, j int);

set gp_workfile_compress_algorithm ='zlib';
set statement_mem='10MB';

create or replace function FuncA()
returns void as
$body$
begin
 	insert into t values(2387283, 'a');
 	insert into t1 values(1, 2);
    CREATE TEMP table TMP_Q_QR_INSTM_ANL_01 WITH(APPENDONLY=true,COMPRESSLEVEL=5,ORIENTATION=row,COMPRESSTYPE=zlib) on commit drop as
    SELECT t1.i from t as t1 join t as t2 on t1.i = t2.i;
EXCEPTION WHEN others THEN
 -- do nothing
end
$body$ language plpgsql;

-- Inject fault before we close workfile in ExecHashJoinNewBatch
\! gpfaultinjector -f workfile_hashjoin_failure -y reset --seg_dbid 2
\! gpfaultinjector -f workfile_hashjoin_failure -y error --seg_dbid 2

select FuncA();
select * from t1;

\! gpfaultinjector -f workfile_hashjoin_failure -y status --seg_dbid 2

drop function FuncA();
drop table t;
drop table t1;

\! gpfaultinjector -f workfile_hashjoin_failure -y reset --seg_dbid 2
