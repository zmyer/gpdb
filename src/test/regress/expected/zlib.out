DROP TABLE IF EXISTS test_zlib_hashjoin;
NOTICE:  table "test_zlib_hashjoin" does not exist, skipping
CREATE TABLE test_zlib_hashjoin (i1 int, i2 int, i3 int, i4 int, i5 int, i6 int, i7 int, i8 int) WITH (APPENDONLY=true) DISTRIBUTED BY (i1) ; 
INSERT INTO test_zlib_hashjoin SELECT i,i,i,i,i,i,i,i FROM 
	(select generate_series(1, nsegments * 333333) as i from 
	(select count(*) as nsegments from gp_segment_configuration where role='p' and content >= 0) foo) bar;
SET gp_workfile_type_hashjoin=bfz;
SET gp_workfile_compress_algorithm=zlib;
SET statement_mem=5000;
--Fail after workfile creation and before add it to workfile set
\! gpfaultinjector -f workfile_creation_failure -y reset --seg_dbid 2
20170125:11:06:19:002833 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_creation_failure -y reset --seg_dbid 2
20170125:11:06:19:002833 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:19:002833 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
\! gpfaultinjector -f workfile_creation_failure -y error --seg_dbid 2
20170125:11:06:19:002845 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_creation_failure -y error --seg_dbid 2
20170125:11:06:19:002845 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:19:002845 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
SELECT COUNT(t1.*) FROM test_zlib_hashjoin AS t1, test_zlib_hashjoin AS t2 WHERE t1.i1=t2.i2;
ERROR:  fault triggered, fault name:'workfile_creation_failure' fault type:'error'  (seg0 slice2 127.0.0.1:25432 pid=2829)
\! gpfaultinjector -f workfile_creation_failure -y status --seg_dbid 2
20170125:11:06:20:002864 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_creation_failure -y status --seg_dbid 2
20170125:11:06:20:002864 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:20:002864 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-fault name:'workfile_creation_failure' fault type:'error' ddl statement:'' database name:'' table name:'' occurrence:'1' sleep time:'10' fault injection state:'completed'  num times hit:'1' 
20170125:11:06:20:002864 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
RESET statement_mem;
DROP TABLE IF EXISTS test_zlib_hagg; 
NOTICE:  table "test_zlib_hagg" does not exist, skipping
CREATE TABLE test_zlib_hagg (i1 int, i2 int, i3 int, i4 int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i1' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO test_zlib_hagg SELECT i,i,i,i FROM 
	(select generate_series(1, nsegments * 300000) as i from 
	(select count(*) as nsegments from gp_segment_configuration where role='p' and content >= 0) foo) bar;
SET statement_mem=2000;
--Fail after workfile creation and before add it to workfile set
\! gpfaultinjector -f workfile_creation_failure -y reset --seg_dbid 2
20170125:11:06:21:002876 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_creation_failure -y reset --seg_dbid 2
20170125:11:06:21:002876 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:21:002876 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
\! gpfaultinjector -f workfile_creation_failure -y error --seg_dbid 2
20170125:11:06:21:002891 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_creation_failure -y error --seg_dbid 2
20170125:11:06:21:002891 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:21:002891 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
SELECT MAX(i1) FROM test_zlib_hagg GROUP BY i2;
ERROR:  fault triggered, fault name:'workfile_creation_failure' fault type:'error'  (seg0 slice2 127.0.0.1:25432 pid=2829)
\! gpfaultinjector -f workfile_creation_failure -y status --seg_dbid 2
20170125:11:06:22:002907 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_creation_failure -y status --seg_dbid 2
20170125:11:06:22:002907 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:22:002907 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-fault name:'workfile_creation_failure' fault type:'error' ddl statement:'' database name:'' table name:'' occurrence:'1' sleep time:'10' fault injection state:'completed'  num times hit:'1' 
20170125:11:06:22:002907 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
-- Reset faultinjectors
\! gpfaultinjector -f workfile_creation_failure -y reset --seg_dbid 2
20170125:11:06:22:002919 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_creation_failure -y reset --seg_dbid 2
20170125:11:06:22:002919 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:22:002919 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
create table t (i int, j text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into t select i, i from generate_series(1,1000000) as i;
create table t1(i int, j int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
set gp_workfile_compress_algorithm ='zlib';
set statement_mem='10MB';
create or replace function FuncA()
returns void as
$body$
begin
 	insert into t values(2387283, 'a');
 	insert into t1 values(1, 2);
    CREATE TEMP table TMP_Q_QR_INSTM_ANL_01 WITH(APPENDONLY=true,COMPRESSLEVEL=5,ORIENTATION=row,COMPRESSTYPE=zlib) on commit drop as
    SELECT t1.i from t as t1 join t as t2 on t1.i = t2.i;
EXCEPTION WHEN others THEN
 -- do nothing
end
$body$ language plpgsql;
-- Inject fault before we close workfile in ExecHashJoinNewBatch
\! gpfaultinjector -f workfile_hashjoin_failure -y reset --seg_dbid 2
20170125:11:06:24:002931 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_hashjoin_failure -y reset --seg_dbid 2
20170125:11:06:24:002931 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:24:002931 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
\! gpfaultinjector -f workfile_hashjoin_failure -y error --seg_dbid 2
20170125:11:06:24:002943 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_hashjoin_failure -y error --seg_dbid 2
20170125:11:06:24:002943 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:24:002943 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
select FuncA();
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CONTEXT:  SQL statement "CREATE TEMP table TMP_Q_QR_INSTM_ANL_01 WITH(APPENDONLY=true,COMPRESSLEVEL=5,ORIENTATION=row,COMPRESSTYPE=zlib) on commit drop as SELECT t1.i from t as t1 join t as t2 on t1.i = t2.i"
PL/pgSQL function "funca" line 4 at SQL statement
 funca 
-------
 
(1 row)

select * from t1;
 i | j 
---+---
(0 rows)

\! gpfaultinjector -f workfile_hashjoin_failure -y status --seg_dbid 2
20170125:11:06:25:002955 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_hashjoin_failure -y status --seg_dbid 2
20170125:11:06:25:002955 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:25:002955 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-fault name:'workfile_hashjoin_failure' fault type:'error' ddl statement:'' database name:'' table name:'' occurrence:'1' sleep time:'10' fault injection state:'completed'  num times hit:'1' 
20170125:11:06:25:002955 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
drop function FuncA();
drop table t;
drop table t1;
\! gpfaultinjector -f workfile_hashjoin_failure -y reset --seg_dbid 2
20170125:11:06:26:002967 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f workfile_hashjoin_failure -y reset --seg_dbid 2
20170125:11:06:26:002967 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on content=0:dbid=2:mode=s:status=u
20170125:11:06:26:002967 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
